pkg/controller/operators/catalogtemplate/operator.go

```diff
@@ -130,6 +130,13 @@ func (o *Operator) syncCatalogSources(obj interface{}) error {
 		return nil
 	}
 
+	// This catalog template feature is not usable in configmap/internal catalogsource
+	// TODO: Filter the cache on catalogsource tyle (perhaps via labels)
+	if inputCatalogSource.Spec.SourceType == v1alpha1.SourceTypeInternal || inputCatalogSource.Spec.SourceType == v1alpha1.SourceTypeConfigmap {
+		o.logger.Debugf("skipping unsupported catalogsource type(s): %#v", obj)
```

Comment:
Do we need to scale down this information to just the obj.GetName() or obj.GetNamespace()? I was playing around with this locally and here's the log that was printed:

```
time="2021-11-17T19:10:07Z" level=debug msg="skipping unsupported catalogsource type(s): &v1alpha1.CatalogSource{TypeMeta:v1.TypeMeta{Kind:\"\", APIVersion:\"\"}, ObjectMeta:v1.ObjectMeta{Name:\"mock-ocs-main-update2-jzxmb\", GenerateName:\"\", Namespace:\"operators\", SelfLink:\"\", UID:\"3e2d6fc6-0308-40ab-abca-53a030551c6f\", ResourceVersion:\"915\", Generation:1, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:63772772981, loc:(*time.Location)(0xa6e2680)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil), ClusterName:\"\", ManagedFields:[]v1.ManagedFieldsEntry{v1.ManagedFieldsEntry{Manager:\"e2e-local.test\", Operation:\"Update\", APIVersion:\"operators.coreos.com/v1alpha1\", Time:(*v1.Time)(0xb0f6cd8), FieldsType:\"FieldsV1\", FieldsV1:(*v1.FieldsV1)(0xba31140), Subresource:\"\"}, v1.ManagedFieldsEntry{Manager:\"catalog\", Operation:\"Update\", APIVersion:\"operators.coreos.com/v1alpha1\", Time:(*v1.Time)(0xb0f6cf0), FieldsType:\"FieldsV1\", FieldsV1:(*v1.FieldsV1)(0xba31150), Subresource:\"\"}}}, Spec:v1alpha1.CatalogSourceSpec{SourceType:\"internal\", Priority:0, ConfigMap:\"mock-ocs-main-update2-jzxmb-configmap\", Address:\"\", Image:\"\", UpdateStrategy:(*v1alpha1.UpdateStrategy)(nil), Secrets:[]string(nil), DisplayName:\"\", Description:\"\", Publisher:\"\", Icon:v1alpha1.Icon{Data:\"\", MediaType:\"\"}}, Status:v1alpha1.CatalogSourceStatus{Message:\"\", Reason:\"\", LatestImageRegistryPoll:(*v1.Time)(nil), ConfigMapResource:(*v1alpha1.ConfigMapResourceReference)(0xb9a7780), RegistryServiceStatus:(*v1alpha1.RegistryServiceStatus)(0xb9a7800), GRPCConnectionState:(*v1alpha1.GRPCConnectionState)(0xbaac8a0), Conditions:[]v1.Condition(nil)}}"
time="2021-11-17T19:10:08Z" level=error msg="UpdateStatus - error while setting CatalogSource status" error="Operation cannot be fulfilled on catalogsources.operators.coreos.com \"mock-ocs-main-update2-jzxmb\": the object has been modified; please apply your changes to the latest version and try again" id=EvQSV source=mock-ocs-main-update2-jzxmb
```

Reply:
Oh right. I realized that before as well. Haven't had a chance to update it.